src/components/common/SceneWrapper.tsx

```diff
@@ -372,3 +375,38 @@ const SceneFooter = styled(View)({
   left: 0,
   right: 0
 })
+
+/**
+ * A hacky component to fix floating nav-bar issues caused by the keyboard being
+ * opened before a navigation is dispatched.
+ *
+ * This component listens for the 'beforeRemove' event on the navigation object
+ * and prevents the default behavior of leaving the screen if the keyboard is
+ * visible. It dismisses the keyboard and waits for the 'keyboardWillHide'
+ * event before dispatching the navigation action.
+ *
+ * @returns null
+ */
+function FloatingNavFixer() {
+  const navigation = useNavigation()
```

Comment:
We generally pass `navigation` as a prop instead of using the `useNavigation` hook. It's not safe to put `useNavigation` inside an Airship modal, and we don't want components to secretly blow up if you use them in the wrong place.